src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -332,6 +332,90 @@ vs. ``it('image edit test' () => ...)``. As shown, the first example makes gramm
 prefix whereas the second message does not.``it()`` should describe what the individual test is doing and
 what it expects to happen.
 
+Differences between Frontend Unit Tests and End-to-End (E2E) Tests / FAQ
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+General introduction about testing and E2E/unit tests 
+
+
+What are E2E/unit tests designed for?
+.....................................
+
+E2E test:
+
+"Protractor is an end-to-end test framework for Angular and AngularJS applications.
+Protractor runs tests against your application running in a real browser,
+interacting with it as a user would." `(src) <http://www.protractortest.org/#/>`__
+
+It requires a fully functional system and tests the interaction of all components
+of the application (Ceph, back-end, front-end).
+E2E tests are designed to mimic the behavior of the user when interacting with the application
+- for example when it comes to workflows like creating/editing/deleting an item.
+Also the tests should verify that certain items are displayed as a user would see them
+when clicking through the UI (for example a menu entry or a pool that has been
+created during a test and the pool and its properties should be displayed in the table).
+
+Angular Unit Tests:
+
+Unit tests, as the name suggests, are tests for smaller units of the code.
+Those tests are designed for testing all kinds of Angulars' components (e.g. services, pipes etc.).
+They do not require a connection to the backend and hence are independent of it,
+although the expected result of the backend can be verified using only the frontend.
```

Comment:
Could you please have a look at the sentence again? - It was odd to read for me. But maybe it's just me, then it's fine ;)

Reply:
What do you think of the following?

They do not require a connection to the backend, hence those tests are independent of it. The expected data of the backend is mocked in the frontend and by using this data the functionality of the frontend can be tested without having to have real data from the backend.

Reply:
Jep, that's perfect. Thank you :)

Reply:
Thanks for the feedback, @s0nea. Updated this section accordingly.